ansible/modules/hashivault/hashivault_init.py

```diff
@@ -63,22 +67,70 @@
 
 def main():
     argspec = hashivault_argspec()
-    argspec['secret_shares'] = dict(required=False, type='int', default=5)
-    argspec['secret_threshold'] = dict(required=False, type='int', default=3)
+    argspec['seal_method'] = dict(
+        required=False, type='str', default='secret', choices=['secret', 'recovery']
+    )
+    argspec['secret_shares'] = dict(required=False, type='int', default=None)
+    argspec['secret_threshold'] = dict(required=False, type='int', default=None)
     argspec['pgp_keys'] = dict(required=False, type='list', default=None)
     argspec['root_token_pgp_key'] = dict(required=False, type='str', default=None)
     argspec['stored_shares'] = dict(required=False, type='int', default=None)
     argspec['recovery_shares'] = dict(required=False, type='int', default=None)
     argspec['recovery_threshold'] = dict(required=False, type='int', default=None)
     argspec['recovery_pgp_keys'] = dict(required=False, type='list', default=None)
     module = hashivault_init(argspec)
+    additional_parameter_handling(module)
     result = hashivault_initialize(module.params)
     if result.get('failed'):
         module.fail_json(**result)
     else:
         module.exit_json(**result)
 
 
+def additional_parameter_handling(module):
+    '''Additional parameter validation'''
+
+    # Default shares and thresholds depending on the seal method
+    if module.params.get('seal_method') == 'secret':
+        module.params['secret_shares'] = (
+            5
+            if module.params.get('secret_shares') is None
```

Comment:
Why bother with the None check instead of defaulting in the get. Isn't params just a dict so you could module.params.get('secret_shares', 5)?

Reply:
I did that at first but it doesn't work.

```
        module.params['recovery_shares'] = module.params.get('recovery_shares', 5)

        module.fail_json(
            msg=f"Here's why {module.params['recovery_shares']}",
            changed=False,
        )
```
![image](https://github.com/TerryHowe/ansible-modules-hashivault/assets/2412909/b407baa9-2a82-4108-a148-09f06201e2e6)

Reply:
My guess is that since `default=None` the key actually exists in the dict and `get()` must check the nonexistence of the key for the default value to be returned

Reply:
![image](https://github.com/TerryHowe/ansible-modules-hashivault/assets/2412909/f9950259-ba2e-477a-bd52-71e433e6a96a)

Reply:
@TerryHowe Does that work for you?